tests/rag_answers/custom_deepeval/metrics/test_factual_correctness.py

```diff
@@ -0,0 +1,580 @@
+            )
+
+            # let _calculate_score and _finalise_evaluation run naturally
+            result = await metric.a_measure(test_case)
+
+            assert round(result, 4) == round(expected_score, 4)
+            assert metric.success == expected_success
+
+        @pytest.mark.asyncio
+        async def test_native_model_sets_evaluation_costs(
+            self,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+            fact_classification_result: FactClassificationResult,
+        ):
+            expected_cost = 0.05
+
+            mock_native_model.a_generate = AsyncMock(
+                return_value=(fact_classification_result, expected_cost)
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_native_model)
+
+            _ = await metric.a_measure(test_case)
+
+            assert metric.evaluation_cost == expected_cost
+            mock_native_model.a_generate.assert_awaited_once()
+
+        @pytest.mark.asyncio
+        @patch(
+            "govuk_chat_evaluation.rag_answers.custom_deepeval.metrics.factual_correctness.factual_correctness.initialize_model"
+        )
+        async def test_non_native_model_returns_valid_schema(
+            self,
+            mock_initialize_model,
+            mock_non_native_model: Mock,
+            test_case: LLMTestCase,
+            fact_classification_result: FactClassificationResult,
+        ):
+            # force initialize_model to return (mock_model, False)
+            mock_initialize_model.return_value = (mock_non_native_model, False)
+
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)  # type: ignore
+            result = await metric.a_measure(test_case)
+
+            assert isinstance(metric.confusion_matrix, ClassifiedFacts)
+            assert (
+                metric.confusion_matrix == fact_classification_result.classified_facts
+            )
+            assert isinstance(result, float)
+            assert result == metric.score
+
+        @pytest.mark.asyncio
+        async def test_non_native_model_empty_confusion_matrix_triggers_error(
+            self,
+            mock_non_native_model: Mock,
+            test_case,
+            caplog: pytest.LogCaptureFixture,
+        ):
+            mock_non_native_model.a_generate = AsyncMock(
+                return_value=FactClassificationResult(
+                    classified_facts=ClassifiedFacts()
+                )
+            )
+
+            caplog.set_level(logging.ERROR)
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)
+
+            result = await metric.a_measure(test_case)
+
+            assert mock_non_native_model.a_generate.call_count == 1
+
+            captured_logs = caplog.text
+            assert (
+                "Error: no facts were classified. confusion_matrix is empty for input:"
+                in captured_logs
+            )
+
+            assert math.isnan(result)
```

Comment:
I think we should also assert here that metric.error is populated with what we expect